File: BEmuJava/bemu/src/main/java/com/bemu/BEmu/IntradayTickRequest/RequestIntradayTickElementBool.java
//------------------------------------------------------------------------------
// <copyright project="BEmuJava" file="BEmu.IntradayTickRequest.RequestIntradayTickElementBool.java" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------

package com.bemu.BEmu.IntradayTickRequest;

public class RequestIntradayTickElementBool extends RequestIntradayTickElementString
{
    private final boolean _value;

    RequestIntradayTickElementBool(String elementName, boolean value)
    {
    	super(elementName, String.valueOf(value).toLowerCase());
        this._value = value;
    }

    //I can't override GetElementAsBool here because the Bloomberg Request object stores bools as strings, not bools.  You can't convert the string to a bool
    boolean getBoolean()
    {
		return this._value; 
    } 
}

File: BEmu_cpp/src/IntradayTickRequest/RequestIntradayTickElementBool.cpp
//------------------------------------------------------------------------------------------------
// <copyright project="BEmu_cpp" file="src/IntradayTickRequest/RequestIntradayTickElementBool.cpp" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------------------------

#include "bemu_headers.h"
#include "IntradayTickRequest/RequestIntradayTickElementBool.h"
#include "IntradayTickRequest/RequestIntradayTickElementString.h"
#include "BloombergTypes/Name.h"
#include "BloombergTypes/Element.h"
#include "Types/IndentType.h"

namespace BEmu
{
	namespace IntradayTickRequest
	{
		RequestIntradayTickElementBool::RequestIntradayTickElementBool(std::string& elementName, bool bvalue) : RequestIntradayTickElementString(elementName, bvalue ? "True" : "False")
		{
			this->_bvalue = bvalue;
		}

		bool RequestIntradayTickElementBool::getBool()
		{
			return this->_bvalue;
		}
	}
}